OpenSR/scripts/server/orders/AttackOrder.as

```diff
@@ -126,10 +139,48 @@ tidy class AttackOrder : Order {
 			return OS_COMPLETED;
 		}
 
-		//Set effector targets
-		ship.blueprint.target(obj, target, flags);
+		// Not sure what this does with ships, but just do nothing if
+		// attacking with something else
+		if (ship !is null) {
+			//Set effector targets
+			ship.blueprint.target(obj, target, flags);
+		}
```

Comment:
So I think I've started figuring out what non-ships are supposed to do here...

The `addWeapon()` function in `obj_selection` iterates through an object's hooks (though it looks like it doesn't account for building hooks - not sure) to look for effectors whose weapon ranges should be displayed. The same mechanism could be used to get the `Turret`s associated with an object and tell them to target the target.

This would allow orbital/planetary weapons to be aimed at the order's target.

There's a catch, though. Turret targeting is usually either left to work... somehow... or handled via a ship blueprint. In order to make this work, it may be necessary to reimplement large swaths of the blueprint's targeting code in AngelScript, where we can manipulate the Turrets ourselves.

Reply:
The only thing in the game that can to turn to shoot is a ship right? Planets that can move and orbitals that can move don't have the ability to target, since they automatically spin? Hence I don't think there needs to be any additional code here to handle non ships.

Reply:
Yeaaaah, except that's not what that does. Planets and orbitals may have 360-degree firing arcs on all their weapons, but this is not the same as not targeting things.

Targeting is relevant for two reasons:
1) Effectors have tracking speeds. To my knowledge, the turret cannot instantaneously rotate to face a target, even if it's not attached to a ship blueprint.
2) Even if turrets *could* instantly turn towards a target (which is technically true for missiles, sort of, since they're guided projectiles), they need to know what target they're turning towards.